CC/Sounder/receiver.cc

```diff
@@ -497,16 +520,32 @@ void Receiver::loopRecv(int tid, int core_id, SampleBuffer* rx_buffer) {
           if (config_->dl_data_slot_present() == true) {
             while (-1 != baseTxData(radio_id, cell, frame_id, rxTimeBs))
               ;
-            this->notifyPacket(kBS, frame_id + this->txFrameDelta, 0, radio_id,
+            this->notifyPacket(kBS, frame_id + this->txFrameDelta_, 0, radio_id,
                                bs_tx_buff_size);  // Notify new frame
           } else {
             this->baseTxBeacon(radio_id, cell, frame_id,
-                               rxTimeBs + txTimeDelta);
+                               rxTimeBs + txTimeDelta_);
           }  // end if config_->dul_data_slot_present()
         }
         if (!config_->isPilot(frame_id, slot_id) &&
-            !config_->isUlData(frame_id, slot_id))
+            !config_->isUlData(frame_id, slot_id)) {
+          for (size_t ch = 0; ch < num_packets; ++ch) {
+            const int bit = 1 << (cursor + ch) % sizeof(std::atomic_int);
+            //                std::cout<<"bit at 2 is " << bit << std::endl;
+            const int offs = (cursor + ch) / sizeof(std::atomic_int);
+            const int old =
+                std::atomic_fetch_and(&pkt_buf_inuse[offs], ~bit);  // now full
+            //                std::cout<<"old at 2 is " << old << std::endl;
+            // if buffer was full, exit
+            if ((old & bit) != bit) {
+              MLPD_ERROR("thread %d buffer full\n", tid);
+              throw std::runtime_error("Thread %d buffer full\n");
+            }
+            // Reserved until marked empty by consumer
```

Comment:
remove comment as it is not accurate

Reply:
Fix the comments (or use this code chunk)
      for (size_t ch = 0; ch < num_packets; ++ch) {
        const int bit = 1 << (cursor + ch) % sizeof(std::atomic_int);
        const int offs = (cursor + ch) / sizeof(std::atomic_int);
        const int old = std::atomic_fetch_and(&pkt_buf_inuse[offs], ~bit);  // now empty
        // if buffer was empty, exit
        if ((old & bit) != bit) {
          MLPD_ERROR("thread %d freed buffer when already free\n", tid);
          throw std::runtime_error("buffer empty during free");
        }
      }

Reply:
thanks for the comment